### PR TITLE
Add profile contributions section

### PR DESCRIPTION
**Pull request by iHiD**
Screenshot 2021-04-14 at 22 56 23
Screenshot 2021-04-14 at 23 10 58

### CODE REVIEW

File: app/helpers/react_components/profile/contributions_summary.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
+module ReactComponents
+  module Profile
+    class ContributionsSummary < ReactComponent
+      # We normally proxy this to the current renderng session
```
Comment (ErikSchierboom) [inline app/helpers/react_components/profile/contributions_summary.rb]:
```suggestion
      # We normally proxy this to the current rendering session
```

File: app/helpers/react_components/profile/contributions_summary.rb [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+      memoize
+      def mentored_students
+        user.mentor_discussions.joins(solution: :exercise).group('exercises.track_id').
+          select('solutions.user_id').distinct.count
+      end
+
+      memoize
+      def reputation_points
+        @user.reputation_tokens.group(:track_id, :category).sum(:value).with_indifferent_access
```
Comment (iHiD) [inline app/helpers/react_components/profile/contributions_summary.rb]:
```suggestion
        @user.reputation_tokens.group(:track_id, :category).sum(:value)
```

File: app/javascript/components/profile/contributions-chart/use-chart.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  gradient.addColorStop(1, `rgba(158, 0, 255, ${opacity})`)
+
+  return gradient
+}
+
+const padReputation = (reputation: number[]): number[] => {
+  const max = reputation.reduce((a, b) => a + b, 0)
+  const min = max / 15
+
+  return reputation.map((r) => (r > min ? r : min))
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-chart/use-chart.ts]:
```suggestion
  return reputation.map((r) => Math.max(r, min))
```

File: app/javascript/components/profile/contributions-chart/use-chart.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      labels: paddedReputation.map(() => ''),
+      datasets: [
+        {
+          label: '',
+          data: paddedReputation,
+          backgroundColor: (context) => {
+            return trackColor
+              ? `rgba(${trackColor}, 0.3)`
+              : createBluePurpleGradient(context.chart, 0.3)
+          },
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-chart/use-chart.ts]: You could get rid of the curly braces and the return I think:

```suggestion
          backgroundColor: (context) => 
            trackColor
              ? `rgba(${trackColor}, 0.3)`
              : createBluePurpleGradient(context.chart, 0.3)
          ,
```

File: app/javascript/components/profile/contributions-chart/use-chart.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+              ? `rgba(${trackColor}, 1)`
+              : createBluePurpleGradient(context.chart, 1)
+          },
+          pointRadius: 5,
+          pointHoverRadius: 5,
+        },
+      ],
+    }
+  }, [JSON.stringify(paddedReputation)])
+  const config: ChartConfiguration<'radar'> = useMemo(() => {
```
Comment (ErikSchierboom) [inline app/javascript/components/profile/contributions-chart/use-chart.ts]: Maybe add a newline here?
```suggestion

  const config: ChartConfiguration<'radar'> = useMemo(() => {
```

File: test/helpers/react_components/profile/contributions_summary_test.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+              { id: :publishing, reputation: 0, metric: "2 solutions" },
+              # TODO: Reputation
+              { id: :mentoring, reputation: 0, metric: "4 students" },
+              { id: :authoring, reputation: 60, metric: "6 exercises" },
+              { id: :building, reputation: 24, metric: "2 PRs created" },
+              { id: :maintaining, reputation: 15, metric: "3 PRs reviewed" },
+              { id: :other, reputation: 0 }
+            ]
+          }
+
```
Comment (ErikSchierboom) [inline test/helpers/react_components/profile/contributions_summary_test.rb:83]:
```suggestion
```

**Review by ErikSchierboom — APPROVED** [review]

**Comment by iHiD** [comment]
@kntsoriano Great start!! I've added the LHS designs too for you.
- All the categories should show on the LHS in the same order as around the box. 
- If there is no reputation it should say "No rep" not "0 rep"

Screenshot 2021-04-15 at 18 18 29

File: app/javascript/components/profile/contributions-chart/use-chart.ts (hunk 3 above)
Comment (kntsoriano) [reply app/javascript/components/profile/contributions-chart/use-chart.ts]: Thanks!

**Pull request merged**